trade-finance-manager-api/src/v1/controllers/acbs.controller.js

```diff
@@ -150,25 +155,32 @@ const checkAzureAcbsFunction = async () => {
   }
 };
 
+/**
+ * Issues facilities in the ACBS, if prerequisites are satisfied.
+ * 1. Ensure facility has been issued in Portal.
+ * 2. Facility has been created in ACBS.
+ * 3. Facility in ACBS is `06` stage.
+ * @param {object} deal - The deal object containing information about the deal, including facilities and ACBS details.
+ * @returns {Promise<Object>} - A promise that resolves with the results of adding the facilities to ACBS log.
+ */
 const issueAcbsFacilities = async (deal) => {
-  if (!deal.tfm || !deal.tfm.acbs) {
+  if (!deal?.tfm?.acbs) {
     /**
      * A facility can only be issued (if unissued),
      * the deal has been submitted to ACBS and acknowledged by the TFM.
      * If the above is false, please do not proceed.
      */
+    console.error('Unable to issue facility in ACBS for deal %s', deal._id);
     return false;
   }
+  
   /**
-   * ACBS verification has been removed due to an ongoing bug of not receiving
-   * the `acbs` object imperative data thus preventing maker from issuing the facility.
-   * TO-DO:
-   * !isIssued(facilityStageInAcbs) && !facility.tfm.acbs.issuedFacilityMaster
-   * const facilityStageInAcbs = facility.tfm.acbs && facility.tfm.acbs.facilityStage;
+   * ACBS verification: Facility stage verification ensures facility which are
+   * under commitment stage (06) are only eligible for issuance
    */
-
+  console.error('Issuing facilities in ACBS for deal %s', deal._id);
   const acbsIssuedFacilitiesPromises = deal.facilities
-    .filter((facility) => facility.hasBeenIssued)
+    .filter((facility) => facility?.hasBeenIssued && !isIssuedInACBS(facility?.tfm?.acbs.facilityStage))
```

Comment:
Sorry -- This is my lack of understanding with this as a general business concept -- if the facility has been issued but the tfm object has not been issued in acbs then it's issued?